x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/spatial/LuceneComponent2DUtils.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.scalar.spatial;
+
+import org.apache.lucene.geo.Component2D;
+import org.apache.lucene.geo.LatLonGeometry;
+import org.apache.lucene.geo.Rectangle;
+import org.apache.lucene.geo.XYGeometry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This utilities class provides access to protected methods in Lucene using alternative APIs.
+ * For example, the 'create' method returns the original Component2D array, instead of a Component2D containing
+ * a component tree of potentially multiple components. This is particularly useful for algorithms that need to
+ * operate on each component individually.
+ */
+public class LuceneComponent2DUtils {
+    /**
+     * This method is based on LatLonGeometry.create, but returns an array of Component2D objects for multi-component geometries.
+     */
+    public static Component2D[] createLatLonComponents(LatLonGeometry... latLonGeometries) {
```

Comment:
Move towards InvalidArgumentException since those have proper HTTP codes.

Reply:
Right now, all the spatial functions use IllegalArgumentException, and that was based on copying from other functions. I think I should move the whole lot over to InvalidArgumentException in one go, in a separate PR, if that is OK for you?